WMF Framework/Widget/WidgetController.swift

```diff
@@ -167,3 +176,96 @@ public final class WidgetController: NSObject {
     #endif
     
 }
+
+/// When the old widget data loading model is removed, this should be moved out of this extension into the class itself and refactored (e.g. the properties here don't need to be computed).
+public extension WidgetController {
+
+    // MARK: - Properties
+
+    fileprivate var widgetCacheDirectoryContainerURL: URL {
+        FileManager.default.wmf_containerURL()
+    }
+
+    fileprivate var widgetCacheDataFileURL: URL {
+        return widgetCacheDirectoryContainerURL.appendingPathComponent("Widget Cache").appendingPathExtension("json")
+    }
+
+    // MARK: - Widget Cache
+
+    /// Returns currently cached `WidgetCache` or an empty default
+    func loadCache() -> WidgetCache {
+        if let data = try? Data(contentsOf: widgetCacheDataFileURL), let decodedCache = try? JSONDecoder().decode(WidgetCache.self, from: data) {
+            return decodedCache
+        }
+
+        return WidgetCache(settings: .default, featuredContent: nil)
+    }
+
+    func saveCache(_ widgetCache: WidgetCache) {
+        let encoder = JSONEncoder()
+        guard let encodedCache = try? encoder.encode(widgetCache) else {
+            return
+        }
+
+        try? encodedCache.write(to: widgetCacheDataFileURL)
+    }
+
+    func updateCacheWith(featuredContent: WidgetFeaturedContent) {
```

Comment:
I notice this method isn't called, is the plan to (eventually) call it from the app-side Explore feed update?

Reply:
Yeah exactly – it's a holdover from when I was implementing that app-side feed updating, but pulled it out last minute.

Would you prefer I remove this function until we properly implement that?

Reply:
@staykids nah that's okay, it can stay. Might be a good to add a comment above explaining that's the future plan.